### PR TITLE
auth: api - Check DNSNames that should be hostnames

### LINKED ISSUE #6636: PowerDNS API doesn't correctly check mx records format
- Program: Authoritative
 - Issue type: Bug report

### Short description

PowerDNS **API** doesn't throw any errors when insert a MX record like this:

`mydomain.tld. MX 1 mail@mydomain.tld.`

Via cli, pdnsutils shows the warning:

```
$ pdnsutil check-zone mydomain.tld
[Warning] MX record in zone 'mydomain.tld' has non-hostname content 'mail@mydomain.tld.'.
```

### Environment
 - Operating system: Ubuntu 16.04
 - Software version: 4.1.1-1pdns.xenial
 - Software source: pdns-server, pdns-backend-mysql 

### Steps to reproduce

1. Add this RRSET: `mydomain.tld. MX 1 mail@mydomain.tld.`

### Expected behaviour
Throw an error like: `MX record in zone 'mydomain.tld' has non-hostname content 'mail@mydomain.tld.'.`

### Actual behaviour
Record Saved

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
This is a copy of the check done in `pdnsutil check-zone` that will trigger an API error when trying to create or update an MX record with a non hostname content.

This fix #6636 

### Checklist

I have:
- [ ] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [x] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -1709,6 +1709,31 @@ static void patchZone(HttpRequest* req, HttpResponse* resp) {
...
+              } else if (rr.qtype.getCode() == QType::MX) {
+                vector<string> parts;
+                stringtok(parts, rr.getZoneRepresentation());
+                if (parts.size() == 2) toCheck = DNSName(parts[1]);
+              } else {
+                toCheck = DNSName(rr.content);
+              }
+
+              if (toCheck.empty()) {
+                throw ApiException("RRset "+qname.toString()+" IN "+qtype.getName() + " unable to extract hostname from content.");
```
Comment (pieterlexis) [inline pdns/ws-auth.cc]: Perhaps "Bad Request" is a better error here?

Comment (chbruyand) [reply pdns/ws-auth.cc]: AFAIK the rest of the API will return 400 only when the parsing of the JSON failed. I would prefer to stick to the "Unprocessable entity" error, but can change it of course!

Comment (zeha) [reply pdns/ws-auth.cc]: Please stick to 422 :)

**Comment by chbruyand** [comment]
Rebased and refactored to avoid code duplication.

**Comment by zeha** [comment]
<3 @chbruyand

**Pull request merged**